kv/kv.go

```diff
@@ -119,26 +119,51 @@ const (
 	PresumeKeyNotExists
 )
 
-// Transaction defines the interface for operations inside a Transaction.
-// This is not thread safe.
-type Transaction interface {
-	// Get gets the value for key k from KV store.
+// MemStore is an abstract KV storage.
+type MemStore interface {
```

Comment:
yeah, MemStore may be not a good choice. I thought of KVStore in the first place, but `kv.KVStore` seems verbose. Another choice is using just `Store`, but we have already got a `Store`. Call for some discussions. @qiuyesuifeng @siddontang